zdm/zdm.py

```diff
@@ -89,14 +108,23 @@ def calc_dV(self):
 	def EF(self,alpha=0,bandwidth=1e9):
 		"""Calculates the fluence--energy conversion factors as a function of redshift
 		"""
-		self.FtoE=cos.F_to_E(1,self.zvals,alpha=alpha,bandwidth=bandwidth)
+		if self.alpha_method==0:
+			self.FtoE=cos.F_to_E(1,self.zvals,alpha=alpha,bandwidth=bandwidth)
+		elif self.alpha_method==1:
+			self.FtoE=cos.F_to_E(1,self.zvals,alpha=0.,bandwidth=bandwidth)
+		else:
+			raise ValueError("alpha method must be 0 or 1, not ",self.alpha_method)
 	
-	def set_evolution(self,n):
+	def set_evolution(self,n,alpha=None):
 		""" Scales volumetric rate by SFR """
 		self.sfr_n=n
-		self.sfr=cos.sfr(self.zvals)**n
-	
-	
+		if alpha is not None:
+			self.alpha=alpha
+		#self.sfr=cos.sfr(self.zvals)**n #old hard-coded value
+		self.sfr=self.source_evolution(self.zvals,n)
+		if self.alpha_method==1:
+			self.sfr *= (1.+self.zvals)**(-self.alpha) #reduces rate with alpha
```

Comment:
This is for the rate interpretation of alpha right? Why is the self.sfr being multiplied with (1+z)^-𝛼, when in the paper it states that "Under the rate interpretation, the FRB population itself behaves as Φ ∼ Φ(𝑧, 𝜈) = Φ(𝑧)𝜈^𝛼, and the k-correction therefore directly changes the rate, adding an additional factor of (1 + 𝑧)^𝛼 to Eq. 12 (which is Φ(𝑧) ={Φ0/(1 + z)}{SFR(𝑧)/SFR(0)}^n.

Reply:
Hi Esan - because the original definition in the paper used F(nu)~nu^-alpha. Then it was suggested to change the definition in the paper, but I did not dare tinker with the code at that point. It's a to-do.

Reply:
Okay, got it.